fix(learn): Broke several steps down into smaller steps for the Cafe Menu project - v7

RandellDawson opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x]  My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

This PR adds most of the suggestions @moT01 made on PR #39407 that were not implemented before merging.  There are changes to many steps' descriptions/instructions.  Also, an additional 7 steps were added to make the project steps a bit more granular.

RandellDawson committed:
fix: made some verbiage changes to existing steps

RandellDawson committed:
fix: rewrite step 45 instructions

RandellDawson committed:
fix: added two steps between 45 and 46

RandellDawson committed:
fix: miscellaneous edits for steps 21,22, 27

RandellDawson committed:
fix: change step 35 to allow for smaller steps

RandellDawson committed:
fix: update 36-41 seed code

RandellDawson committed:
fix: added new step between 41 and 42

RandellDawson committed:
fix: add new step between 42 and 43

RandellDawson committed:
fix: broke step 33 into 4 new steps

RandellDawson committed:
fix: removed unnecessary step 37

RandellDawson committed:
fix: change wording of step 33 instructions

RandellDawson committed:
fix: corrected ERM syntax

RandellDawson committed:
fix: corrected seed code typos in steps 45 and 46

moT01 reviewed (changes requested):
I found a few typos. There's an issue around lesson 53, I dug around a little and the mongoid is repeated on 55. So when I go through it, lesson 55 doesn't show up, and lesson 53 seems like it's supposed to be where 55 is - and the actual lesson 53 is missing I think.

If lesson 53 adds the classes needed after 52, I think these are broken up nicely.

Part 16 and 17 - moving the styles to the css file: The preview does not behave like described. We have them remove the <style> tags and code on one lesson, then link to the stylesheet on the next. But the styles do not get removed from the preview when deleting the code - and it says that it does. Not sure can actually remove the styles when the code is in the css. If we can't find a way to remove the styles, we could possibly link to the stylesheet first, then remove the styles.

scissorsneedfoodtoo commented on curriculum/challenges/english/01-responsive-web-design/basic-css-cafe-menu/part-069.md:
```diff
@@ -8,7 +8,10 @@ isHidden: true
 <section id='description'>
 
-Using the same style selector in the previous step, make the font size of the items and prices by using a value of `18px`.
+The default properties of an `hr` element will make it appear as a thin light grey line. You can change the heighth of the line by specifying a value for the `height` property.
```
The default properties of an `hr` element will make it appear as a thin light grey line. You can change the height of the line by specifying a value for the `height` property.

scissorsneedfoodtoo commented on curriculum/challenges/english/01-responsive-web-design/basic-css-cafe-menu/part-076.md:
```diff
@@ -8,9 +8,7 @@ isHidden: true
-You change properties of a link when the mouse hovers them by using a <dfn>pseudo-selector</dfn> that looks like `a:hover { propertyName: propertyValue }`
-
-Change the color of the footer `Visit our website` link to be `brown` when a user hovers over it.
+Using the same style selector in the previous step, make the font size of the items and prices by using a value of `18px`.
```
Using the same style selector in the previous step, make the font size of the items and prices larger by using a value of `18px`.

scissorsneedfoodtoo commented on curriculum/challenges/english/01-responsive-web-design/basic-css-cafe-menu/part-077.md:
```diff
@@ -8,10 +8,9 @@ isHidden: true
-You change properties of a link when the link is actually clicked by using a <dfn>pseudo-selector</dfn> that looks like `a:active { propertyName: propertyValue }`
-
-Change the color of the footer `Visit our website` link to be `white` when a clicks on it.
+Changing the `bottom-margin` to `5px` looks great. However, now the space between the `Cinaommon Roll` menu item and the second `hr` element does not match the space between the top `hr` element and the `Coffees` heading.
```
Changing the `bottom-margin` to `5px` looks great. However, now the space between the `Cinnamon Roll` menu item and the second `hr` element does not match the space between the top `hr` element and the `Coffees` heading.

scissorsneedfoodtoo commented on curriculum/challenges/english/01-responsive-web-design/basic-css-cafe-menu/part-087.md:
```diff
@@ -0,0 +1,190 @@
+## Description
+<section id='description'>
+
+To remove some of the vertical space between the `h1` element and the text `Est. 2020`, change the bottom margin of the `h1` to `15px`;
```
To remove some of the vertical space between the `h1` element and the text `Est. 2020`, change the bottom margin of the `h1` to `15px`.

scissorsneedfoodtoo commented on curriculum/challenges/english/01-responsive-web-design/basic-css-cafe-menu/part-046.md:
````diff
@@ -8,14 +8,7 @@ isHidden: true
-Cheesecake 3.00
-Cinammon Roll
-```
+To complete the styling, add the applicable class names `flavor` and `price` to all the remaining `p` elements.
````
To complete the styling, add the class names `flavor` and `price` to all the remaining `p` elements.

RandellDawson replied:
The reason I had the original wording instead of what you propose, is I did not want the user to think they should add both class names to all the remaining `p` elements.  Any suggestion on how to accomplish that?

scissorsneedfoodtoo replied:
@RandellDawson, ah, I see what you mean. I was more focused on simplifying, but I think you're right -- the original wording makes it more clear that the class names `flavor` and `price` should only be added to certain paragraph elements.

I don't think there's a better way to say that, so let's leave it as is.

scissorsneedfoodtoo reviewed (changes requested):
Thanks for catching those things @moT01. I noticed a few more typos which I left as review comments.

Also I tried to break up some paragraphs so the instructional text stands out a bit better. It wasn't always possible, but I left suggestions where I could.

Another thing I noticed is that there are some instances where there are two spaces after a period. Those get stripped out before being rendered to the page, but it would be good to clean them up here.

Finally, this might be a little nit picky, but maybe we can consider changing the "Coffees" and "Desserts" headers to "Coffee" and "Dessert". The use of "coffees" and "desserts" in the descriptions seem fine, but we might want to reword things to avoid the whole countable/uncountable noun issue.

For example, `part-008` could read, "...It will eventually contain pricing information about the types of coffee and dessert offered by the cafe."

scissorsneedfoodtoo commented on curriculum/challenges/english/01-responsive-web-design/basic-css-cafe-menu/part-082.md:
```diff
@@ -8,7 +8,9 @@ isHidden: true
 <section id='description'>
 
-Now apply the `address` class to the `p` element containing the address.
+You change properties of a link when the link has actually been visited by using a <dfn>pseudo-selector</dfn> that looks like `a:visited { propertyName: propertyValue }`
```
You change properties of a link when the link has actually been visited by using a <dfn>pseudo-selector</dfn> that looks like `a:visited { propertyName: propertyValue }`.

scissorsneedfoodtoo commented on curriculum/challenges/english/01-responsive-web-design/basic-css-cafe-menu/part-083.md:
```diff
@@ -8,9 +8,9 @@ isHidden: true
 <section id='description'>
 
-The menu looks good, but other than the coffee beans background image, it is mainly just text.
+You change properties of a link when the mouse hovers them by using a <dfn>pseudo-selector</dfn> that looks like `a:hover { propertyName: propertyValue }`
```
You change properties of a link when the mouse hovers them by using a <dfn>pseudo-selector</dfn> that looks like `a:hover { propertyName: propertyValue }`.

scissorsneedfoodtoo commented on curriculum/challenges/english/01-responsive-web-design/basic-css-cafe-menu/part-084.md:
```diff
@@ -8,9 +8,10 @@ isHidden: true
 <section id='description'>
 
-The image you added is not centered horizontally like the `Coffees` heading above it.  `img` elements are "like" inline elements.
+You change properties of a link when the link is actually clicked by using a <dfn>pseudo-selector</dfn> that looks like `a:active { propertyName: propertyValue }`
```
You change properties of a link when the link is actually clicked by using a <dfn>pseudo-selector</dfn> that looks like `a:active { propertyName: propertyValue }`.

RandellDawson committed:
fix: corrected typo

Co-authored-by: Tom <20648924+moT01@users.noreply.github.com>

RandellDawson committed:
fix: changed wording to improve readability

Co-authored-by: Kristofer Koishigawa <scissorsneedfoodtoo@gmail.com>

RandellDawson committed:
fix: changed "Free Code Camp" to "freeCodeCamp"

Co-authored-by: Tom <20648924+moT01@users.noreply.github.com>

RandellDawson committed:
fix: corrected typo and improved wording

Co-authored-by: Tom <20648924+moT01@users.noreply.github.com>

RandellDawson committed:
fix: corrected typo

Co-authored-by: Tom <20648924+moT01@users.noreply.github.com>

RandellDawson committed:
fix: correct typos

Co-authored-by: Kristofer Koishigawa <scissorsneedfoodtoo@gmail.com>

RandellDawson committed:
fix: corrected punctuation

Co-authored-by: Kristofer Koishigawa <scissorsneedfoodtoo@gmail.com>

RandellDawson committed:
fix: corrected punctuation

Co-authored-by: Kristofer Koishigawa <scissorsneedfoodtoo@gmail.com>

RandellDawson committed:
fix: corrected punctuation

Co-authored-by: Kristofer Koishigawa <scissorsneedfoodtoo@gmail.com>

RandellDawson committed:
fix: added missing period

Co-authored-by: Kristofer Koishigawa <scissorsneedfoodtoo@gmail.com>

RandellDawson committed:
fix: corrected typo

Co-authored-by: Kristofer Koishigawa <scissorsneedfoodtoo@gmail.com>

RandellDawson committed:
fix: added the missing word "larger"

Co-authored-by: Kristofer Koishigawa <scissorsneedfoodtoo@gmail.com>

RandellDawson committed:
fix: corrected typo

Co-authored-by: Kristofer Koishigawa <scissorsneedfoodtoo@gmail.com>

RandellDawson commented on curriculum/challenges/english/01-responsive-web-design/basic-css-cafe-menu/part-059.md:
```diff
@@ -8,7 +8,9 @@ isHidden: true
-Inside the `footer`, add a `p` element. Then, nest an anchor (`a`) element in the `p` that links to `https://www.freecodecamp.org` and has the text `Visit our website`.
+You can change the `font-family` of text, to make it look different the the default font of your browser. Each browser has some common fonts available to it.
+
+Change all the text in your `body`, by adding a `font-family` property with the value `sans-serif`.  This is a fairly common font that is very readable.
```
@scissorsneedfoodtoo How about the following?

>Change all the text in your body by adding a font-family property with the value sans-serif. This is a fairly common font available to most browsers that is very readable.

scissorsneedfoodtoo replied:
@RandellDawson, yes, that looks good. The only change I suggested was removing the comma after `body`. Otherwise I think everything looks good.

RandellDawson committed:
fix: change paragraph to `p`

RandellDawson committed:
fix: reorderd steps after step 37 deleted

RandellDawson committed:
fix: added two new steps to fix missing steps problem

RandellDawson commented:
@moT01 @scissorsneedfoodtoo I believe I have addressed all the concerns via commits except the one below by @scissorsneedfoodtoo 

> To complete the styling, add the class names flavor and price to all the remaining p elements.

Since we decided to leave the one paragraph instructionsfor now, please let me know if I need to do anything else with this PR.

scissorsneedfoodtoo commented:
@RandellDawson, thanks for going through all those comments. I went ahead and resolved the "To complete the styling, add the class names..." suggestion because I don't think there's a better way to communicate that.

I just left a couple of the other suggestions open. They're pretty minor, though, and I'm fine with merging this as-is.

RandellDawson committed:
fix: added missing ERMs for step 69

moT01 reviewed (approved):
LGTM @RandellDawson 🎉 This is coming along nicely 🎉
